src/main/kotlin/com/sourcegraph/cody/chat/ui/ContextFilesPanel.kt

```diff
@@ -40,67 +43,148 @@ class ContextFilesPanel(
   }
 
   @RequiresBackgroundThread
-  private fun updateFileList(contextFileNames: Set<String>) {
+  private fun updateFileList(
+      localFiles: Map<LocalContextFile, ContextFile>,
+      remoteFiles: Map<RemoteContextFile, ContextFile>
```

Comment:
I would say lets discuss this offline, I think we have too many different files abstractions already.
`ContextFile`, `ContextFileFile`, `LocalContextFile`, `RemoteContextFile`, `VirtualFile`, nio `Path`, `URI`s,  and what not.
We will get lost in that pretty quick, I think we need to do some simplifications there.